test/OpenTelemetry.Tests/Metrics/MetricExemplarTests.cs

```diff
@@ -64,55 +68,206 @@ public void TestExemplarsCounter(MetricReaderTemporalityPreference temporality)
+            DateTime testStartTime,
+            List<Metric> exportedItems,
+            (double Value, bool ExpectTraceId)[] firstMeasurementValues,
+            (double Value, bool ExpectTraceId)[] secondMeasurementValues,
+            Func<Exemplar, double> getExemplarValueFunc)
         {
-            Assert.Single(exemplars);
+            var metricPoint = GetFirstMetricPoint(exportedItems.Where(m => m.Name == instrumentName));
+
+            Assert.NotNull(metricPoint);
+            Assert.True(metricPoint.Value.StartTime >= testStartTime);
+            Assert.True(metricPoint.Value.EndTime != default);
+
+            var exemplars = GetExemplars(metricPoint.Value);
+
+            if (temporality == MetricReaderTemporalityPreference.Cumulative)
+            {
+                // Current design:
+                //  First collect we saw Exemplar A & B
+                //  Second collect we saw Exemplar C but B remained in the reservoir
+                Assert.Equal(2, exemplars.Count);
+                secondMeasurementValues = secondMeasurementValues.Concat(firstMeasurementValues.Skip(1).Take(1)).ToArray();
+            }
+            else
+            {
+                Assert.Single(exemplars);
+            }
+
+            ValidateExemplars(exemplars, metricPoint.Value.StartTime, metricPoint.Value.EndTime, secondMeasurementValues, getExemplarValueFunc);
         }
+    }
 
-        ValidateExemplars(exemplars, metricPoint.Value.StartTime, metricPoint.Value.EndTime, secondMeasurementValues);
+    [Theory]
+    [InlineData(MetricReaderTemporalityPreference.Cumulative)]
+    [InlineData(MetricReaderTemporalityPreference.Delta)]
+    public void TestExemplarsObservable(MetricReaderTemporalityPreference temporality)
+    {
+        DateTime testStartTime = DateTime.UtcNow;
+        var exportedItems = new List<Metric>();
+
+        (double Value, bool ExpectTraceId)[] measurementValues = new (double Value, bool ExpectTraceId)[]
+        {
+            (18D, false),
+            (19D, false),
+        };
+
+        int measurementIndex = 0;
+
+        using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+        var gaugeDouble = meter.CreateObservableGauge("testGaugeDouble", () => measurementValues[measurementIndex].Value);
+        var gaugeLong = meter.CreateObservableGauge("testGaugeLong", () => (long)measurementValues[measurementIndex].Value);
+        var counterDouble = meter.CreateObservableCounter("counterDouble", () => measurementValues[measurementIndex].Value);
+        var counterLong = meter.CreateObservableCounter("counterLong", () => (long)measurementValues[measurementIndex].Value);
+
+        using var container = this.BuildMeterProvider(out var meterProvider, builder => builder
+            .AddMeter(meter.Name)
+            .SetExemplarFilter(new AlwaysOnExemplarFilter())
+            .AddView(i =>
+            {
+                if (i.Name.StartsWith("testGauge"))
```

Comment:
i am not able to see the need for custom sized (3) for this instrument, and not others? Are we validating them separately?

Reply:
Good catch. I removed the view from this test. Probably got there copy/pasting 🤣 Observables are an interesting case for exemplars. We only observe them during collect so there will only ever be at most 1 exemplar exported regardless of reservoir size. If user is doing `TraceBasedExemplarFilter` then they likely will _never_ see an exemplar for observables because they are collected outside of all traces unless you go and force a trace in the observe callback which feels very strange. Might need to do something about that in SDK/spec?

Reply:
yes. I want to see if spec would change the defaults, so that exemplars are opt-in. specifically, users *must* opt-in to exemplars for each instrument. (The key reasoning being - most users just need exemplar for one key instrument, typically the one measuring latency. The current model enables Exemplars for every instrument, including Observables as well, leading to wasted cpu/memory etc.)